### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by fabriziosestito**
This PR is the first iteration of Trento telemetry,
it adds a simple endpoint to collect hosts telemetry data and a storage adapter abstraction to store the same data in more databases.
It also implements the influxdb (v2) storage adapter.

Let's start from here!

### CODE REVIEW

File: server/influxdb.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package server
+
+import (
+	"context"
+
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+)
+
+const (
+	org                      = "trento"
```
Comment (arbulu89) [inline server/influxdb.go:10]: Most probably these values shouldn't be constants, at least the `org` and `bucket`.
I would make them come from cmd or env variables.

Anyway, easy to change in the future!

File: server/influxdb.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	return &InfluxDB{url: url, token: token}
+}
+
+func (i *InfluxDB) StoreHostTelemetry(h *HostTelemetry) error {
+	client := i.getClient()
+	writeAPI := client.WriteAPIBlocking(org, bucket)
+	defer client.Close()
+
+	p := influxdb2.NewPointWithMeasurement(hostTelemetryMeasurement).
+		AddTag("agent_id", h.AgentID).
```
Comment (arbulu89) [inline server/influxdb.go:30]: As an initial version is good enough. We will refine this if needed

File: server/server.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+//go:generate mockery --name=StorageAdapter --inpackage --filename=storage_adapter_mock.go
+type StorageAdapter interface {
+	StoreHostTelemetry(*HostTelemetry) error
+}
+
+type HostTelemetry struct {
+	AgentID       string    `json:"agent_id"`
+	SLESVersion   string    `json:"sles_version"`
+	CPUCount      int       `json:"payload"`
```
Comment (arbulu89) [inline server/server.go]: Is this `payload` a typo?

File: server/server.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	SocketCount   int       `json:"socket_count"`
+	TotalMemoryMB int       `json:"total_memory_mb"`
+	CloudProvider string    `json:"cloud_provider"`
+	Time          time.Time `json:"time"`
+}
+
+func hostTelemetryHandler(adapters ...StorageAdapter) func(w http.ResponseWriter, r *http.Request) {
```
Comment (arbulu89) [inline server/server.go:27]: Maybe using `gin` would be better, to keep consistency. Let's leave it for a future discussion hehe

File: server/server.go [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		var hostTelemetry HostTelemetry
+		err := json.NewDecoder(r.Body).Decode(&hostTelemetry)
+		if err != nil {
+			log.Errorf("Error unmarshaling host telemetry: %v", err)
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		log.Debugf("Received new host telemetry data from agent: %s", hostTelemetry.AgentID)
+		for _, adapter := range adapters {
```
Comment (arbulu89) [inline server/server.go:43]: Why do we need multiple `adapters` here?
I don't see the use case right now as we are only storing host telemetry

**Review by arbulu89 — APPROVED** [review]
Fantasitic @fabriziosestito ,
I think as a 1st version is more than enough.
In fact, I would appreciate if we dockerize the app as soon as possible, as this will unblock the deployment part.

File: server/server.go (hunk 5 above)
Comment (fabriziosestito) [reply server/server.go:43]: @arbulu89 we are going to add the postgresql adapter in the following PRs, we still store the host telemetry but in another database

File: server/server.go (hunk 3 above)
Comment (fabriziosestito) [reply server/server.go]: It is, thanks!

File: server/server.go (hunk 4 above)
Comment (fabriziosestito) [reply server/server.go:27]: Yep, I just used the standard library since it was very minimal. I'll keep this on the radar.

**Review by dottorblaster — APPROVED** [review]
LGTM

**Review by nelsonkopliku — APPROVED** [review]
Excellent starting point! Thanks!

**Pull request merged**